### PR TITLE
[Relax][PyTorch] Fix masked_select VM build

### PR DESCRIPTION
**Pull request by V-aerus**
This PR fixes the PyTorch ExportedProgram importer lowering for `torch.masked_select`.

  Previously, `masked_select` lowered to:

  - flatten data and mask
  - `nonzero(mask_flat)`
  - `squeeze(axis=[0])`
  - `take(data_flat, indices)`

  However, the result of `R.nonzero` only carried rank information. The following `R.squeeze` over the dynamic nonzero output could remain unhandled during build/VM execution.

  This PR inserts a `match_cast` after `R.nonzero` using the exported output metadata, preserving the dynamic selected-length dimension before `squeeze`.

  A numerical regression test is also added to cover:

  PyTorch eager -> torch.export -> Relax import -> build -> VM run -> output comparison

  Testing:

  - `python -m pytest -q tests/python/relax/test_frontend_from_exported_program.py -k 'masked_select'`

### CODE REVIEW

File: python/tvm/relax/frontend/torch/base_fx_graph_translator.py [hunk 1]
```diff
@@ -2619,6 +2619,16 @@ def _masked_select(self, node: fx.Node) -> relax.Var:
...
+        tensor_meta = node.meta.get("tensor_meta")
+        if tensor_meta is not None and len(tensor_meta.shape) == 1:
+            num_selected = tensor_meta.shape[0]
+            if not isinstance(num_selected, int):
+                num_selected = tirx.Var(str(num_selected), "int64")
+        else:
+            num_selected = tirx.Var(f"{node.name}_num_selected", "int64")
+        indices = self.block_builder.match_cast(
+            indices, relax.TensorType([1, num_selected], "int64")
+        )
```
[the remark this answers is not in the document]
Comment (V-aerus) [reply python/tvm/relax/frontend/torch/base_fx_graph_translator.py:2631]: Thanks for the review. I checked the current TVM Relax API and existing code patterns. `BlockBuilder.match_cast` is used with `relax.TensorType(shape, dtype)` in the current codebase, for example in `tests/python/relax/test_blockbuilder_core.py`, and the ONNX frontend has a similar `nonzero ->
  match_cast` pattern using `relax.TensorType([1, num_nonzero], "int64")`.

  Also, in this TVM version, `relax.TensorType` accepts a shape and dtype. I don't see `relax.TensorStructInfo` being used in the current tree. The current code passes the targeted masked_select tests, so I will keep this as `TensorType` unless maintainers prefer a different API.

**Review by tlopex — APPROVED** [review]
LGTM Thanks
